libnavigation-core/src/main/java/com/mapbox/navigation/core/RoutesInvalidatedObserver.kt

```diff
@@ -0,0 +1,22 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.route.NavigationRoute
+
+/**
+ * Observer that is notified of routes being invalidated.
+ */
+fun interface RoutesInvalidatedObserver {
+
+    /**
+     * Invoked when routes are invalidated.
+     * It means that specified routes won't be refreshed anymore:
+     * you won't receive traffic updates, incident updates, etc.
+     * It is recommended to rebuild a route once it becomes invalidated.
+     * Note: if route A becomes invalidated and you continue using it instead of rebuilding it,
+     * you won't be notified the second time that this route is no longer valid:
+     * you'll only be notified once about a specific route being invalidated.
+     *
+     * @param routes list of routes that became invalidated
+     */
+    fun onRoutesInvalidated(routes: List<NavigationRoute>)
```

Comment:
I noticed that we tend to add more parameters to callbacks later as find more corner cases and lear more about feature in general. Maybe it makes sense to make all the callbacks extendable by design?
```suggestion
    class InvalidatedRoutesParams internal constructor(val invalidatedRoutes: List<NavigationRoutes>) 

    fun onRoutesInvalidated(params: InvalidatedRoutesParams)
```